README.md

```diff
@@ -25,39 +27,64 @@ const Fastify = require('fastify')
 const server = Fastify()
 
 server.register(require('fastify-http-proxy'), {
-  upstream,
-  prefix: '/upstream', // optional
+  upstream: 'http://my-api.example.com',
+  prefix: '/api', // optional
+  http2: false // optional
+})
+
+server.listen(3000)
+```
+
+This will proxy any request starting with `/api` to `http://my-api.example.com`. For instance `http://localhost:3000/api/users` will be proxied to `http://my-api.example.com/users`.
+
+If you want to have different proxies on different prefixes in you can register multiple instances of the plugin as shown in the following snippet:
+
+```js
+const Fastify = require('fastify')
+const server = Fastify()
+const proxy = require('fastify-http-proxy')
+
+server.register(proxy, {
+  upstream: 'http://my-api.example.com',
+  prefix: '/api', // optional
+  http2: false // optional
+})
+
+server.register(proxy, {
+  upstream: 'http://single-signon.example.com/auth',
+  prefix: '/auth', // optional
   http2: false // optional
 })
 
 server.listen(3000)
 ```
 
-For a more complete example, see `example.js`.
+Notice that in these case it is important to use the `prefix` option and to have *non-ambiguous* or *non-overlapping* prefixes (e.g. `/api` and `/api2`).
```

Comment:
I think the non-overlapping part is a bug.

Reply:
@lmammino what happens if you use `/api` and `/api2`?

Reply:
I am actually not sure how prefixes are resolved and I haven't tested if this specific example is ambiguous.

Anyway, we could actually make this a little bit better by saying something like this:

> Notice that in these case it is important to use the `prefix` option and to have *non-ambiguous* or *non-overlapping* prefixes (e.g. don't use `/api` and `/api2` but `/api/` and `/api2/`).

I am assuming here the second case will behave properly, but maybe it's worth updating the tests to validate these cases.

What do you think @mcollina ? Any opinion from you @delvedor ?

Reply:
I am going to add a test and refactor the description if no issue is raised

Reply:
https://github.com/fastify/fastify-http-proxy/pull/11 :)